Feature/edc viewer desktop

flo7842 opened the pull request.

flo7842 committed:
Adding viewer with electron app

flo7842 committed:
set JavaDoc, remove method, rename enum

flo7842 committed:
Method for define the path of executable.


Update Readme

flo7842 committed:
Merge branch 'master' into feature/edc_viewer_desktop

flo7842 committed:
open url

flo7842 committed:
Run app with url POST request

flo7842 committed:
Merge branch 'master' into feature/edc_viewer_desktop

greg7859 commented on src/main/java/fr/techad/edc/popover/utils/OpenUrlAction.java:
```diff
@@ -42,19 +94,14 @@ public void openUrl(String url) throws IOException, URISyntaxException {
+            if(helpConfiguration.getViewerDesktopPath().isEmpty()){
+                LOGGER.error("The path of the application must be entered");
+            }else {
+                setTimeout(() -> System.out.println("test"), 12000, url);
```
I don't undestand the goal of this method: wait 12s, display text and fetch an url.

greg7859 commented on src/main/java/fr/techad/edc/popover/utils/OpenUrlAction.java:
```diff
@@ -42,19 +94,14 @@ public void openUrl(String url) throws IOException, URISyntaxException {
+            }else {
+                setTimeout(() -> System.out.println("test"), 12000, url);
+
+                    Runtime.getRuntime().exec(helpConfiguration.getViewerDesktopPath(), null, null);
```
use the Process to know if the viewer is running.
See waitFor() behavior to change the state of the viewer.

flo7842 committed:
Adding method to run Electron desktop viewer

flo7842 committed:
Set method to check if process is running




Change interval

flo7842 committed:
Merge branch 'master' into feature/edc_viewer_desktop

greg7859 commented on src/main/java/fr/techad/edc/popover/internal/swing/builder/ContextualContentComponentBuilderImpl.java:
```diff
@@ -235,6 +235,8 @@ private void openUrl(String url) {
             LOGGER.error("Error on IO", e);
         } catch (URISyntaxException e) {
             LOGGER.error("Impossible to open the browser with url:{}", url);
+        } catch (Exception e) {
```
Perhaps you can focus on the target exception and not with the root exception

greg7859 commented on src/main/java/fr/techad/edc/popover/internal/swing/builder/ContextualContentComponentBuilderImpl.java:
```diff
@@ -235,6 +235,8 @@ private void openUrl(String url) {
         } catch (URISyntaxException e) {
             LOGGER.error("Impossible to open the browser with url:{}", url);
+        } catch (Exception e) {
+            e.printStackTrace();
```
never use printStackTrace, use a logger with debug level

greg7859 commented on src/main/java/fr/techad/edc/popover/internal/swing/components/IconButtonListener.java:
```diff
@@ -109,6 +111,8 @@ private void openBrowser() {
             LOGGER.error("Impossible to open the browser with url:{}", url);
         } catch (IOException e) {
             LOGGER.error("Error on IO", e);
+        } catch (Exception e) {
```
same as previously

greg7859 commented on src/main/java/fr/techad/edc/popover/utils/OpenUrlAction.java:
```diff
@@ -8,27 +8,103 @@
         super();
         this.browser = browser;
         this.helpConfiguration = helpConfiguration;
+
```
remove this empt line

greg7859 commented on src/main/java/fr/techad/edc/popover/utils/OpenUrlAction.java:
```diff
@@ -8,27 +8,103 @@
+    }
+
+    private HttpURLConnection postViewerURL(String url) throws IOException {
+        String viewerUrl = "http://localhost:60000/viewerurl";
```
and if the viewser works on different port, you have to enable the configuration of this url.

greg7859 commented on src/main/java/fr/techad/edc/popover/utils/OpenUrlAction.java:
```diff
@@ -38,23 +114,19 @@ public OpenUrlAction(Browser browser, HelpConfiguration helpConfiguration) {
-            catch (IOException e)
-            {
-                LOGGER.error(String.valueOf(e));
+            if(helpConfiguration.getViewerDesktopPath().isEmpty()){
```
Format your source code

greg7859 commented on src/main/java/fr/techad/edc/popover/utils/OpenUrlAction.java:
```diff
@@ -38,23 +114,19 @@ public OpenUrlAction(Browser browser, HelpConfiguration helpConfiguration) {
+                StreamGobbler streamGobbler = new StreamGobbler(edcDesktopProcess.getInputStream(), System.out::println);
+                Executors.newSingleThreadExecutor().submit(streamGobbler);
+
+                setInterval(url);
```
setInterval and you define an url.

Are you sure that the method name is correct?

greg7859 commented on src/main/java/fr/techad/edc/popover/utils/OpenUrlAction.java:
```diff
@@ -8,27 +8,103 @@
+        }
+    }
+
+    private void setInterval(String url){
```
review this method name

greg7859 commented on src/main/java/fr/techad/edc/popover/utils/OpenUrlAction.java:
```diff
@@ -8,27 +8,103 @@
+
+    }
+
+    private HttpURLConnection postViewerURL(String url) throws IOException {
```
Be careful, you manage the lifecycle of the connection in this method. At the end of the request, you close the connection. So some action will be impossible to do.

In your case, you want to get the status code. So return it and avoid to reuse a closed connection.

flo7842 committed:
Set method for URL desktop viewer, resolve correction

greg7859 commented on src/main/java/fr/techad/edc/popover/internal/model/HelpConfigurationImpl.java:
```diff
@@ -39,6 +39,8 @@ public class HelpConfigurationImpl implements HelpConfiguration {
     private Font headerFontAttributes = new Font("Dialog", Font.BOLD, 20);
     private Color titleColor = Color.BLACK;
+    private int viewerDesktopPortURL = 60000;
+    private String viewerDesktopHostURL = "http://localhost:";
```
Declare the url directly: http://localhost:60000

flo7842 committed:
Set method for display Electron desktop viewer

greg7859 commented on src/main/java/fr/techad/edc/popover/model/HelpConfiguration.java:
```diff
@@ -248,11 +254,18 @@ public interface HelpConfiguration {
+     * Return the url of desktop viewer
      *
-     * @return the path
+     * @return the host
```
Return the host or the url?

flo7842 committed:
Renaming the return element of a function in the javadoc

flo7842 committed:
Set error title

greg7859 commented on src/main/java/fr/techad/edc/popover/internal/desktop/EdcDesktopProcess.java:
```diff
@@ -0,0 +1,44 @@
+        BufferedReader input = new BufferedReader(new InputStreamReader(process.getInputStream()));
+
+        if(input != null){
+            Runtime.getRuntime().exec("taskkill /F /IM edc-help-viewer-desktop.exe /T");
```
Works only on windows.
why do you not reuse edcDesktopProcess?

flo7842 committed:
Readme update


Add customization functions

Remove embedded browser functions
Add edc help viewer desktop mode

Refactoring for edc desktop class

greg7859 commented on src/main/java/fr/techad/edc/popover/internal/desktop/EdcDesktopImpl.java:
```diff
@@ -0,0 +1,73 @@
+        }
+    }
+    @Override
+    public void killDesktopProcess() throws IOException {
```
rename the function; shutdownDesktopProcess for example

greg7859 commented on src/main/java/fr/techad/edc/popover/desktop/EdcDesktop.java:
```diff
@@ -0,0 +1,50 @@
+     *
+     * @throws IOException
+     */
+    void killDesktopProcess() throws IOException;
```
rename the function; shutdownDesktopProcess for example

flo7842 committed:
Adding rest apache

flo7842 committed:
Added desktop viewer size config


Added arguments to configure the desktop viewer
